Tests fail with Celery 5.6.3

Several tests started failing with Celery 5.6.3:

```
FAILED t/unit/backends/test_database.py::test_DatabaseBackend::test_backend__pickle_serialization__dict_result__protocol_1 - AttributeError: 'str' object has no attribute 'ignore_result'
FAILED t/unit/backends/test_database.py::test_DatabaseBackend::test_backend__pickle_serialization__str_result__protocol_1 - AttributeError: 'str' object has no attribute 'ignore_result'
FAILED t/unit/backends/test_database.py::test_DatabaseBackend::test_backend__pickle_serialization__bytes_result__protocol_1 - AttributeError: 'str' object has no attribute 'ignore_result'
FAILED t/unit/backends/test_database.py::test_DatabaseBackend::test_backend__json_serialization__dict_result__protocol_1 - AttributeError: 'str' object has no attribute 'ignore_result'
FAILED t/unit/backends/test_database.py::test_DatabaseBackend::test_backend__json_serialization__str_result__protocol_1 - AttributeError: 'str' object has no attribute 'ignore_result'
FAILED t/unit/backends/test_database.py::test_DatabaseBackend::test_backend__task_result_meta_injection - AttributeError: 'str' object has no attribute 'ignore_result'
FAILED t/unit/backends/test_database.py::test_DatabaseBackend::test_secrets__pickle_serialization__protocol_1 - AttributeError: 'str' object has no attribute 'ignore_result'
FAILED t/unit/backends/test_database.py::test_DatabaseBackend::test_secrets__json_serialization__protocol_1 - AttributeError: 'str' object has no attribute 'ignore_result'
```

Example traceback:

```
_ test_DatabaseBackend.test_backend__pickle_serialization__dict_result__protocol_1 _

self = <t.unit.backends.test_database.test_DatabaseBackend object at 0x7f470d916470>

    def test_backend__pickle_serialization__dict_result__protocol_1(self):
        self.app.conf.result_serializer = 'pickle'
        self.app.conf.accept_content = {'pickle', 'json'}
        self.b = DatabaseBackend(app=self.app)

        tid2 = uuid()
>       request = self._create_request(
            task_id=tid2,
            name='my_task',
            args=['a', 1, SomeClass(67)],
            kwargs={'c': 6, 'd': 'e', 'f': SomeClass(89)},
            task_protocol=1,
        )

t/unit/backends/test_database.py:324:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
t/unit/backends/test_database.py:61: in _create_request
    request = Request(context, decoded=True, task=name)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <Request: my_task[e6c90af1-ad6f-4e04-9fba-35bf5f5013ae] None None>
message = <Context: {'headers': {'lang': None, 'task': 'my_task', 'id': 'e6c90af1-ad6f-4e04-9fba-35bf5f5013ae', 'root_id': None,...ss object at 0x7f470d5807d0>}, {'callbacks': None, 'errbacks': None, 'chord': None, 'chain': None}), 'sent_event': {}}>
on_ack = <function noop at 0x7f470f0eb4c0>, hostname = None, eventer = None
app = None, connection_errors = None, request_dict = None, task = 'my_task'
on_reject = <function noop at 0x7f470f0eb4c0>, body = None, headers = None
decoded = True, utc = True
maybe_make_aware = <function maybe_make_aware at 0x7f470ed171a0>
maybe_iso8601 = <function maybe_iso8601 at 0x7f470ed16d40>, opts = {}
timelimit = (None, None), ignore_result = None

    def __init__(self, message, on_ack=noop,
                 hostname=None, eventer=None, app=None,
                 connection_errors=None, request_dict=None,
                 task=None, on_reject=noop, body=None,
                 headers=None, decoded=False, utc=True,
                 maybe_make_aware=maybe_make_aware,
                 maybe_iso8601=maybe_iso8601, **opts):
        self._message = message
        self._request_dict = (message.headers.copy() if headers is None
                              else headers.copy())
        self._body = message.body if body is None else body
        self._app = app
        self._utc = utc
        self._decoded = decoded
        if decoded:
            self._content_type = self._content_encoding = None
        else:
            self._content_type, self._content_encoding = (
                message.content_type, message.content_encoding,
            )
        self.__payload = self._body if self._decoded else message.payload
        self.id = self._request_dict['id']
        self._type = self.name = self._request_dict['task']
        if 'shadow' in self._request_dict:
            self.name = self._request_dict['shadow'] or self.name
        self._root_id = self._request_dict.get('root_id')
        self._parent_id = self._request_dict.get('parent_id')
        timelimit = self._request_dict.get('timelimit', None)
        if timelimit:
            self.time_limits = timelimit
        self._argsrepr = self._request_dict.get('argsrepr', '')
        self._kwargsrepr = self._request_dict.get('kwargsrepr', '')
        self._on_ack = on_ack
        self._on_reject = on_reject
        self._hostname = hostname or gethostname()
        self._eventer = eventer
        self._connection_errors = connection_errors or ()
        self._task = task or self._app.tasks[self._type]
        ignore_result = self._request_dict.get('ignore_result', None)
        if ignore_result is None:
>           ignore_result = self._task.ignore_result
E           AttributeError: 'str' object has no attribute 'ignore_result'

.tox/py313-django52/lib/python3.13/site-packages/celery/worker/request.py:130: AttributeError
```

I bisected this to https://github.com/celery/celery/commit/c3c19c31dc3e21f16d4d85a8ba8401a9223ace09: in other words, passing a string as the `task` argument to `Request` now fails.  I don't know whether this should be considered a regression in Celery or whether it uncovered a bug in the test suite of django-celery-results.  Could you please advise?

--- a/t/unit/backends/test_database.py
+++ b/t/unit/backends/test_database.py
@@ -52,6 +52,7 @@ class test_DatabaseBackend:
             sent_event = {}
         else:
             headers, properties, body, sent_event = msg
+        headers.setdefault('ignore_result', False)
         context = Context(
             headers=headers,
             properties=properties,

